server/fleet/labels.go

```diff
@@ -221,3 +221,31 @@ type LabelIdentsWithScope struct {
 	LabelScope LabelScope
 	ByName     map[string]LabelIdent
 }
+
+// Equal returns whether or not 2 LabelIdentsWithScope pointers point to equivalent values.
+func (l *LabelIdentsWithScope) Equal(other *LabelIdentsWithScope) bool {
+	if l == nil || other == nil {
+		return l == other
+	}
+
+	if l.LabelScope != other.LabelScope {
+		return false
+	}
+
+	if l.ByName == nil && other.ByName == nil {
+		return true
+	}
+
+	for k, v := range l.ByName {
```

Comment:
This will return true if `other` includes more labels than `l` does. We could avoid this issue by asserting the maps are the same length first (vs. trying to do some O(n+m) thing).

Reply:
oh i definitely had a length check and then I refactored and must've deleted it! 🤦